xbmc/utils/ProgressJob.cpp

```diff
@@ -180,10 +180,20 @@ void CProgressJob::MarkFinished()
   if (m_progress != NULL)
   {
     if (m_updateProgress)
+    {
       m_progress->MarkFinished();
+      //We don't own this pointer and it will be deleted after it's marked finished
+      //just set it to nullptr so we don't try to use it again
+      m_progress = nullptr;
+    }
   }
   else if (m_progressDialog != NULL && m_autoClose)
+  {
     m_progressDialog->Close();
+    //We don't own this pointer and it will be deleted after it's marked finished
+    //just set it to nullptr so we don't try to use it again
+    m_progressDialog = nullptr;
```

Comment:
It is causing crashes with Isengard branch because `CProgressDialog::DoModal()` which initializes `m_progressDialog` to the proper value also implements auto-closing.
Obviously we should use `MarkFinished()` there aswell instead of calling `m_progressDialog->Close();` but IMO this is not the right place to set `m_progressDialog` to `nullptr`. The progress dialog still exists and could still be used by the progress job (for whatever reason).